File: backend/internal/models/league.go
package models

type League struct {
	DbModel
	CorrelationId string `json:"correlation_id"  validate:"required"`
	SeasonID      uint   `json:"season_id" validate:"required"`
	Name          string `json:"name" validate:"required"`
	Teams         []Team `json:"teams"  validate:"required"`
}

File: app/src/models/League.ts
import type { DbModel, Team } from './index'
import type { LeagueGuid, SeasonId } from './ids'

export interface League extends DbModel {
  correlation_id: LeagueGuid
  season_id: SeasonId
  name: string
  teams: Array<Team>
}

export type { LeagueId, LeagueGuid } from './ids'
